src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts

```diff
@@ -136,13 +152,126 @@ export function isImage(array: Uint8Array): boolean {
 	);
 }
 
+export class CopyTextProvider implements DocumentPasteEditProvider {
+
+	static readonly id = 'text';
+	static readonly kind = new HierarchicalKind('text.plain');
+
+	readonly id = PasteTextProvider.id;
+	readonly kind = PasteTextProvider.kind;
+
+	readonly copyMimeTypes = [Mimes.text, 'vscode-editor-data'];
+
+	async prepareDocumentPaste(model: ITextModel, ranges: readonly IRange[], dataTransfer: IReadonlyVSDataTransfer, token: CancellationToken): Promise<undefined | IReadonlyVSDataTransfer> {
+		if (model.uri.scheme === ChatInputPart.INPUT_SCHEME) {
+			return;
+		}
+		const customDataTransfer = new VSDataTransfer();
+		const rangesString = JSON.stringify({ ranges: ranges[0], uri: model.uri.toString() });
+		customDataTransfer.append('additional-editor-data', createStringDataTransferItem(rangesString));
```

Comment:
Try to use a full mime type, something like `application/vnd.code.additional-editor-data`